bullet_train-super_load_and_authorize_resource/test/controllers/concerns/bullet_train/loads_and_authorizes_resource_test.rb

```diff
@@ -1,23 +1,87 @@
 require "test_helper"
 
 class BulletTrain::LoadsAndAuthorizeResourceTest < ActiveSupport::TestCase
-  class TestClass
+  class TestControllerClass < ActionController::Base
     include BulletTrain::LoadsAndAuthorizesResource
+
+    attr_accessor :child_object, :current_user, :parent_object, :team
+
+    def self.regex_to_remove_controller_namespace
+    end
+
+    def can?(*args)
+    end
+  end
+
+  module Users
+    class TestControllerClass < ActionController::Base
+      include BulletTrain::LoadsAndAuthorizesResource
+
+      def self.regex_to_remove_controller_namespace
+        /^BulletTrain::LoadsAndAuthorizeResourceTest::/
+      end
+    end
   end
 
-  test "it defines .model_namespace_from_controller_namespace" do
-    assert TestClass.respond_to?(:model_namespace_from_controller_namespace)
+  test "model_namespace_from_controller_namespace returns an array of modules names based on the classes namespace minus regex_to_remove_controller_namespace" do
+    assert_equal ["BulletTrain", "LoadsAndAuthorizeResourceTest"], TestControllerClass.model_namespace_from_controller_namespace
+    assert_equal ["Users"], Users::TestControllerClass.model_namespace_from_controller_namespace
   end
 
   test "it defines .account_load_and_authorize_resource" do
-    assert TestClass.respond_to?(:account_load_and_authorize_resource)
+    assert TestControllerClass.respond_to?(:account_load_and_authorize_resource)
+  end
+
+  test "it defines .regex_to_remove_controller_namespace" do
+    assert TestControllerClass.respond_to?(:regex_to_remove_controller_namespace)
   end
 
   test "it defines #load_team" do
-    assert TestClass.new.respond_to?(:load_team)
+    assert TestControllerClass.new.respond_to?(:load_team)
+  end
+
+  test "#load_team returns nil child_object and parent_object are nil" do
+    subject = TestControllerClass.new
+    subject.child_object = nil
+    subject.parent_object = nil
+
+    assert_nil subject.load_team
```

Comment:
Typically these sorts of `load_` controller `before_action` aren't meant to have a specific return. I think it'd be clearer if we just detected the intended side-effect directly:

```suggestion
    assert_nil subject.instance_variable_get(:@team)
```